torch/distributed/_sharded_tensor/api.py

```diff
@@ -176,6 +177,22 @@ class TensorInitParams(object):
                                  memory_format=torch.contiguous_format,
                                  pin_memory=False))
 
+def _validate_output_tensor_for_rank(my_rank: int, dst_rank: int, dst_tensor: Optional[torch.Tensor]):
+    if dst_rank == my_rank:
+        if not dst_tensor:
+            raise ValueError(
+                "Argument ``dst_tensor`` must be specified on destination rank."
+            )
+        if torch.count_nonzero(dst_tensor):
```

Comment:
I second @pritamdamania here, we should be able to overwrite any tensor by filling it with the gathered content, there might be cases that user want to use a random tensor and fill it with the gathered content (although this might be rare).

Reply:
got it. I will remove this check then. but just to ensure my understanding correctly, I will not zero the tensor in gather() and add the shards to it directly